### ISSUE TITLE
first round classification notes

### ISSUE THREAD

**Issue by sdtaylor** [sdtaylor/PhenocamCNN#1]
Using VGG16 Places model with max global pooling to extract features, size 224x224

Used 20% validation split.
Training was increased to 20k images using random selection, weighted so all classes have roughly equal sample size.

*None of the gbm results are valid because of the shuffle issue*

#### Class counts
field_status_crop
0  emergence  1298
1  growth state  2550
2  tassles/flowering   841
3  senescing   751
4  fully senesed   654
5  harvested/plowed  1932
6  snow   456
7  flooded   213
8  unkown   152

### Model
Used a gbm. Other parameters may produce slightly better results, but a GridSearchCV below showed they are all *really* similar in terms of accuracy.
`GradientBoostingClassifier(n_estimators=100, max_features=50, max_depth=3, min_samples_split=100, min_samples_leaf=10)`

#### Validation data

```
                   precision    recall  f1-score   support

        emergence       0.21      0.30      0.25       265
     growth state       0.20      0.07      0.11       501
tassles/flowering       0.12      0.20      0.15       172
        senescing       0.09      0.11      0.10       154
   fully senesced       0.08      0.11      0.09       132
 harvested/plowed       0.14      0.02      0.03       389
             snow       0.04      0.11      0.06        80
          flooded       0.02      0.05      0.03        41
          unknown       0.01      0.03      0.01        35

      avg / total       0.15      0.11      0.11      1769

class_description  emergence  growth state  tassles/flowering  senescing  fully senesced  harvested/plowed  snow  flooded  unknown                                                                                                                 
emergence                 79            29                 34         20              24                 6    26       18       29
growth state             120            37                 70         68              53                20    65       26       42
tassles/flowering         26            20                 34         12              18                 5    20       13       24
senescing                 27            15                 25         17              16                 4    20       11       19
fully senesced            20            21                 22         18              15                 2    14        6       14
harvested/plowed          73            38                 66         42              47                 7    54       19       43
snow                      18            13                 10          9               4                 4     9        7        6
flooded                    9             4                 10          5               3                 1     4        2        3
unknown                    5             4                  9          4               7                 0     3        2        1

```

#### Training data

```
                   precision    recall  f1-score   support

        emergence       0.35      0.44      0.39      2250
     growth state       0.35      0.43      0.39      2220
tassles/flowering       0.32      0.40      0.36      2273
        senescing       0.43      0.34      0.38      2201
   fully senesced       0.36      0.39      0.38      2224
 harvested/plowed       0.39      0.34      0.37      2238
             snow       0.41      0.36      0.39      2253
          flooded       0.41      0.32      0.36      2152
          unknown       0.42      0.36      0.39      2189

      avg / total       0.38      0.38      0.38     20000

class_description  emergence  growth state  tassles/flowering  senescing  fully senesced  harvested/plowed  snow  flooded  unknown                                                                                                             
emergence                987           218                241        113             165               131   152      111      132
growth state             193           955                217        128             190               137   124      139      137
tassles/flowering        241           218                911        129             201               143   157      127      146
senescing                246           233                243        757             200               151   137      110      124
fully senesced           237           232                219        126             876               161   134      117      122
harvested/plowed         232           224                262        132             202               771   152      121      142
snow                     229           232                233        127             212               137   812      126      145
flooded                  240           229                225        126             207               171   138      680      136
unknown                  209           193                261        121             179               157   159      116      794

```

### CV Grid Search
the following parameter space was search with the above config, nothing was significantly better

`gbm_grid = dict(n_estimators=[100,500,1000], learning_rate=[0.1,0.01,0.001], max_features = [10,50,100], subsample=[0.1,0.5,1.0], max_depth=[3,5,10], min_samples_split=[50,100], min_samples_leaf=[10,20])
`
